finale/src/components/district.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import Township from './township';
+import data from "../data"
+import dl_logo from "../img/download_icon.png"
+
+class District extends Component {
```

Comment:
actually, right now you could try to transfer the react component like this to the real Pure function with hooks which is more elegant.

Reply:
Do you mean doing something similar to **Adding Hooks to Classes with State** from [this article](https://www.digitalocean.com/community/tutorials/five-ways-to-convert-react-class-components-to-functional-components-with-react-hooks) ?

Reply:
Yep, just tried the `useState `

Reply:
Ok will try it to create the logic for expanding the collapsible rows.

Reply:
Hi @flyingant,

Hope you had a great weekend! I created a separate component called ```DistrictHook``` for the purpose of this challenge and used the useState hook to expand and collapsible rows. This component will render when you select ```District``` in the dropdown filter. 

Looking forward to your feedback. 

Best,
Yi

Reply:
Where is it?

Reply:
Hi @flyingant,

You can find the ```DistrictHook``` component [here](https://github.com/yizu100/test-frontend/blob/master/finale/src/components/districtHook.jsx)